### PR TITLE
Update the `TestSwarmLockUnlockCluster` test to be less flakey.

### PR DESCRIPTION
**Pull request by cyli**
This ensures that we wait until the state is not pending in order to check whether a swarm cluster is locked or unlocked, and ensure that we will optionally wait when transitioning from an autolocked state to a non-autolocked state and vice versa.

Would be useful for 1.13.

cc @tonistiigi @aaronlehmann

### CODE REVIEW

File: integration-cli/docker_cli_swarm_test.go [hunk 1]
```diff
@@ -1079,10 +1079,9 @@ func (s *DockerSwarmSuite) TestSwarmLockUnlockCluster(c *check.C) {
 	c.Assert(getNodeStatus(c, d2), checker.Equals, swarm.LocalNodeStateActive)
 
 	// once it's caught up, d2 is set to not be locked
-	c.Assert(d2.Restart(), checker.IsNil)
```
Comment (cyli) [inline integration-cli/docker_cli_swarm_test.go]: should fix https://jenkins.dockerproject.org/job/Docker%20Master/label=fedora23%20&&%20devicemapper/7771/console

File: integration-cli/docker_cli_swarm_test.go [hunk 2]
```diff
@@ -878,12 +878,51 @@ func (s *DockerSwarmSuite) TestDNSConfigUpdate(c *check.C) {
 	c.Assert(strings.TrimSpace(out), checker.Equals, "{[1.2.3.4] [example.com] [timeout:3]}")
 }
 
+// This tries to get the node status, and if it gets "pending", it waits until it's not pending anymore
 func getNodeStatus(c *check.C, d *SwarmDaemon) swarm.LocalNodeState {
-	info, err := d.info()
```
Comment (cyli) [inline integration-cli/docker_cli_swarm_test.go]: An attempt to fix https://jenkins.dockerproject.org/job/Docker%20Master/label=centos72%20&&%20devicemapper/7769/console

File: integration-cli/docker_cli_swarm_test.go [hunk 3]
```diff
@@ -878,12 +878,51 @@ func (s *DockerSwarmSuite) TestDNSConfigUpdate(c *check.C) {
...
-	c.Assert(err, checker.IsNil)
+	var (
+		info swarm.Info
+		err  error
+	)
+	waitAndAssert(c, defaultReconciliationTimeout, func(*check.C) (interface{}, check.CommentInterface) {
```
Comment (tonistiigi) [inline integration-cli/docker_cli_swarm_test.go]: there is already a method `d.checkLocalNodeState` for this

Comment (cyli) [inline integration-cli/docker_cli_swarm_test.go]: Thanks!

File: integration-cli/docker_cli_swarm_test.go [hunk 4]
```diff
@@ -878,12 +878,44 @@ func (s *DockerSwarmSuite) TestDNSConfigUpdate(c *check.C) {
 	c.Assert(strings.TrimSpace(out), checker.Equals, "{[1.2.3.4] [example.com] [timeout:3]}")
 }
 
+// This tries to get the node status, and if it gets "pending", it waits until it's not pending anymore
 func getNodeStatus(c *check.C, d *SwarmDaemon) swarm.LocalNodeState {
+	waitAndAssert(c, defaultReconciliationTimeout, d.checkLocalNodeState, checker.NotEquals, swarm.LocalNodeStatePending)
```
Comment (tophj-ibm) [inline integration-cli/docker_cli_swarm_test.go]: checker.NotEquals is undefined, it should be checker.Not(checker.Equals)

File: integration-cli/docker_cli_swarm_test.go (hunk 1 above)
Comment (aaronlehmann) [inline integration-cli/docker_cli_swarm_test.go]: I'm not sure this is necessary. The root cause looks like a bug in swarmkit. I'm going to submit a PR to fix it.

Comment (aaronlehmann) [inline integration-cli/docker_cli_swarm_test.go]: Yes the pending one

Comment (cyli) [inline integration-cli/docker_cli_swarm_test.go]: @aaronlehmann Ok, will remove the waiting for the state to not be in pending, then.

**Comment by tonistiigi** [comment]
LGTM

**Comment by aaronlehmann** [comment]
LGTM

**Pull request merged**